pkg/migration/runner.go

```diff
@@ -821,20 +835,24 @@ func (r *Runner) checksum(ctx context.Context) error {
 	// - background flushing
 	// - checkpoint thread
 	// - checksum "replaceChunk" DB connections
-	// Handle a case in the tests not having a dbConfig
+	// Handle a case just in the tests not having a dbConfig
 	if r.dbConfig == nil {
 		r.dbConfig = dbconn.NewDBConfig()
 	}
 	r.db.SetMaxOpenConns(r.dbConfig.MaxOpenConnections + 2)
 	var err error
 	for i := range 3 { // try the checksum up to 3 times.
+		if i > 0 {
+			r.checksumWatermark = "" // reset the watermark if we are retrying.
+		}
```

Comment:
Is this intentional? so resume from checksum watermark only works if first resume is successful, otherwise it starts checksum from scratch again?